### PR TITLE
[DOCS] Update the document of HETERO pipeline parallelism

### PR DESCRIPTION
**Pull request by WeldonWangwang**
### Details:
 - Update the document of HETERO pipeline parallelism

### Tickets:
 - *CVS-138199*

### CODE REVIEW

**Review by zhaixuejun1993 — APPROVED** [review]

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 1]
```diff
@@ -18,29 +18,36 @@ Execution via the heterogeneous mode can be divided into two independent steps:
...
 2. Compiling a model to the Heterogeneous device assumes splitting the model to parts, compiling them on the specified devices (via `ov::device::priorities <https://docs.openvino.ai/2024/api/c_cpp_api/structov_1_1device_1_1_priorities.html>`__), and executing them in the Heterogeneous mode. The model is split to subgraphs in accordance with the affinities, where a set of connected operations with the same affinity is to be a dedicated subgraph. Each subgraph is compiled on a dedicated device and multiple `ov::CompiledModel <https://docs.openvino.ai/2024/api/c_cpp_api/classov_1_1_compiled_model.html#doxid-classov-1-1-compiled-model>`__ objects are made, which are connected via automatically allocated intermediate tensors.
+   
+   If you set pipeline parallel (via ``ov::hint::model_distribution_policy``), the model is split into multiple stages, and each stage is assigned to a different device. The output of one stage is fed as input to the next stage.
```
Comment (zhaixuejun1993) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:22]: split -> splitted

Comment (WeldonWangwang) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:22]: Hi, xuejun, seems the past tense of split is still split

**Review by peterchen-intel — APPROVED** [review]

**Review by kblaszczak-intel — APPROVED** [review]
That is nice, thanks for aligning the header underscores :)

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 2]
```diff
@@ -90,9 +100,33 @@ It does not take into account device peculiarities such as the inability to infe
          :language: cpp
          :fragment: [compile_model]
 
+Pipeline parallelism
+------------------------
+
+The pipeline parallelism is set via ``ov::hint::model_distribution_policy``, This mode is an efficient technique to infer large models on multiple devices. The model is split into multiple stages, and each stage is assigned to a different device (``dGPU``, ``iGPU``, ``CPU``, etc.). This mode assign operations to different devices as reasonably as possible, ensuring that different stages can be executed in sequence and minimizing the amount of data transfer between different devices.
```
Comment (songbell) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:106]:
```suggestion
The pipeline parallelism is set via ``ov::hint::model_distribution_policy``. This mode is an efficient technique to infer large models on multiple devices. The model is split into multiple stages, and each stage is assigned to a different device (``dGPU``, ``iGPU``, ``CPU``, etc.). This mode assign operations to different devices as reasonably as possible, ensuring that different stages can be executed in sequence and minimizing the amount of data transfer between different devices.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 3]
```diff
@@ -90,9 +100,33 @@ It does not take into account device peculiarities such as the inability to infe
...
+
+For large models which don’t fit on a single first priority device, model pipeline parallelism is employed where certain parts of the model are placed on different devices to ensure that the device has enough memory to infer these operations, and assign other operations to next priority device.
+
```
Comment (songbell) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:109]: maybe can re-organize this part? kind of confusing of **different devices, and next priority device**

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst (hunk 2 above)
Comment (WeldonWangwang) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:106]: Updated, thanks.

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst (hunk 3 above)
Comment (WeldonWangwang) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:109]: Yes, the description that may cause confusion has been removed.

**Review by songbell — APPROVED** [review]

**Pull request merged**